nextflow.config

```diff
@@ -1,192 +1,276 @@
 /*
- * -------------------------------------------------
- *  nf-core/rnavar Nextflow config file
- * -------------------------------------------------
- * Default config options for all environments.
- */
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    nf-core/rnavar Nextflow config file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Default config options for all compute environments
+----------------------------------------------------------------------------------------
+*/
 
 // Global default params, used in configs
 params {
 
-  // Workflow flags
-  // TODO nf-core: Specify your pipeline's command line flags
-  genome = false
-  input = null
-  input_paths = null
-  single_end = false
-  outdir = './results'
-  publish_dir_mode = 'copy'
-
-  // Boilerplate options
-  multiqc_config = false
-  email = false
-  email_on_fail = false
-  max_multiqc_email_size = 25.MB
-  plaintext_email = false
-  monochrome_logs = false
-  help = false
-  igenomes_base = 's3://ngi-igenomes/igenomes'
-  tracedir = "${params.outdir}/pipeline_info"
-  igenomes_ignore = false
-  custom_config_version = 'master'
-  custom_config_base = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
-  hostnames = false
-  config_profile_name = null
-  config_profile_description = false
-  config_profile_contact = false
-  config_profile_url = false
-  validate_params = true
-  show_hidden_params = false
-  schema_ignore_params = 'genomes,input_paths'
-
-  // Defaults only, expecting to be overwritten
-  max_memory = 128.GB
-  max_cpus = 16
-  max_time = 240.h
+    // Pipeline options
 
-}
+    // Mandatory option
+    input                      = null // sample sheet
+
+    // Genome and reference options
+    genome                     = 'GRCh38'
+    igenomes_base              = 's3://ngi-igenomes/igenomes'
+    igenomes_ignore            = false
+    save_reference             = false
+    save_merged_fastq          = false
+
+    // Sequence read information
+    read_length                = 150 // Required for STAR to build index and align reads
+
+    // Alignment
+    aligner                    = 'star' // Only STAR is currently supported.
+    star_twopass               = true
+    star_ignore_sjdbgtf        = false // Ignore GTF file while creating index or alignment by STAR
+    star_limitBAMsortRAM       = 0
+    star_outBAMsortingBinsN    = 50
+    star_limitOutSJcollapsed   = 1000000
```

Comment:
Are these public-facing variables? the naming is a bit confusing if so

Reply:
Yes, these are public-facing parameters and they are changed when the pipeline runs on a memory-constrained environment.  I know the names are not so great, but I used the same name as STAR documentation so people know what are they changing.

Reply:
ok nice one